packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -5,6 +5,14 @@ import { ResourcePolicy } from './policy';
 import { RotationSchedule, RotationScheduleOptions } from './rotation-schedule';
 import * as secretsmanager from './secretsmanager.generated';
 
+/**
+ * Character Sets which cause problems for other services when used in passwords.
+ */
+export enum ProblemCharacters {
+  AWS_DMS = ' %+:;{}',
```

Comment:
A few things.

1. You're often concatenating these. Would these be better served as `string` constants?

```ts
export class ProblemCharacters {
  public static readonly AWS_DMS = ' %+:;{}';

  public static readonly SHELL = '~`#$&*()|[]{};<>?!\'\\';
}
```

2. Some docs on these would be nice.
3. I see some characters repeat. Does `SecretManager` handle that correctly? Should we de-duplicate them in the construct before generating a CFN template with duplicates?